controllers/vaultdynamicsecret_controller_test.go

```diff
@@ -694,3 +685,152 @@ func Test_computeRotationTime(t *testing.T) {
 		})
 	}
 }
+
+func Test_computeRelativeHorizonWithJitter(t *testing.T) {
+	staticNow := time.Unix(nowFunc().Unix(), 0)
+	defaultNowFunc := func() time.Time { return staticNow }
+
+	tests := []struct {
+		name           string
+		o              *secretsv1beta1.VaultDynamicSecret
+		minHorizon     time.Duration
+		wantMinHorizon time.Duration
+		wantMaxHorizon time.Duration
+		wantInWindow   bool
+	}{
+		{
+			// in rotation, will enqueue with a new time horizon
+			name: "static-creds-in-window",
+			o: &secretsv1beta1.VaultDynamicSecret{
+				Spec: secretsv1beta1.VaultDynamicSecretSpec{
+					AllowStaticCreds: true,
+				},
+				Status: secretsv1beta1.VaultDynamicSecretStatus{
+					StaticCredsMetaData: secretsv1beta1.VaultStaticCredsMetaData{
+						LastVaultRotation: defaultNowFunc().Unix(),
+						TTL:               30,
+					},
+				},
+			},
+			wantMinHorizon: 30 * time.Second,
+			wantMaxHorizon: 30*time.Second + staticCredsMinDurationForJitter,
+			wantInWindow:   true,
+		},
+		{
+			// not in Vault rotation window, will rotate.
+			name: "static-creds-not-in-rotation-window",
+			o: &secretsv1beta1.VaultDynamicSecret{
+				Spec: secretsv1beta1.VaultDynamicSecretSpec{
+					AllowStaticCreds: true,
+				},
+				Status: secretsv1beta1.VaultDynamicSecretStatus{
+					StaticCredsMetaData: secretsv1beta1.VaultStaticCredsMetaData{
+						TTL:               30,
+						LastVaultRotation: defaultNowFunc().Add(-29 * time.Second).Unix(),
+					},
+				},
+			},
+			minHorizon:     1 * time.Second,
+			wantMinHorizon: 1 * time.Second,
+			wantMaxHorizon: time.Duration(1.2 * float64(time.Second)),
+			wantInWindow:   true,
```

Comment:
It took me a bit to realize that `inWindow` for static-creds is the opposite for dynamic creds (it looks like for static `inWindow=false` means time to renew, for dynamic it means not yet time to renew?).

Is it possible to change the compute*Jitter function so that `inWindow` means the same for both static and dynamic? Otherwise It would be good to clarify this in comments on `computeRelativeHorizonWithJitter()`.

Reply:
Would you mind if we punted this change to another PR? I'd like to decouple the sync horizon computation code for the two variants all together.

Reply:
Sure, maybe a TODO in the comments?

Reply:
Added in f15ecd33a48a280dd7d655077d6ff57bb97fe058